xwiki-platform-core/xwiki-platform-model/xwiki-platform-model-api/src/main/java/org/xwiki/model/reference/EntityReference.java

```diff
@@ -297,15 +305,34 @@ protected void setParameter(String name, Serializable value)
             if (this.parameters == null) {
                 this.parameters = new TreeMap<>();
             }
-            this.parameters.put(name, value);
+            if (PARENT_TYPE_PARAMETER.equals(name)) {
+                setParentTypeParameter(value);
+            } else {
+                this.parameters.put(name, value);
+            }
         } else if (parameters != null) {
             this.parameters.remove(name);
-            if (this.parameters.size() == 0) {
+            if (this.parameters.isEmpty()) {
                 this.parameters = null;
             }
         }
     }
 
+    private void setParentTypeParameter(Serializable value)
```

Comment:
Note that I hesitated here between silently ignoring the arguments or throwing an IllegalArgumentException when there was a parent or when the value wasn't an allowed parent type. I decided to go for the silent solution because it's easier to create a reference by using EntityReference#replaceParent (which is what we use in the resolver): the parameter is just automatically ignored.

Reply:
We also perform this kind of silent ignore for the locale parameter.

Reply:
> silent ignore for the locale parameter

Not really, if the format of the locale String is wrong, you will get an `IllegalArgumentException`.

Reply:
Indeed, I improved a bit the method to have more checks. Note that I'm still skipping setting the parameter value is parent != null, but this check remains silent. wdyt @tmortagne ?

Reply:
I'm fine with the silent ignore when `value != null && getParent() == null`.